### PR TITLE
Install the latest version of php/nginx and MariaDB/ProFTPD (LEMP) server on Debian 9.x

### PR DESCRIPTION
**Pull request by eaglepride**
I have read and understood the Contributor's Certificate of Origin available at the end of 
https://raw.githubusercontent.com/hetzneronline/community-content/master/tutorial-template.md
and I hereby certify that I meet the contribution criteria described in it.
Signed-off-by: zhang dawei <139310@gmail.com>

### CODE REVIEW

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/.keep [hunk 1]
```diff
@@ -0,0 +1 @@
+11
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/.keep]: I suppose the .keep file is no longer needed since you've created other files in the directory :)

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+
+
+**Prerequisites**
+
+need a "new" install or "clean" debian .
+
+## Step 1 - (Optional) Disable postfix and remove it 
+
+please ignore this step if your system not installed postfix .
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: I'd rephrase this:
```suggestion
You may ignore this step if your system does not have postfix installed.
```

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+```shell
+systemctl disable postfix && /etc/init.d/postfix stop
+apt remove -y --purge postfix*
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: Maybe we could add a little warning for our readers to be careful when using wildcards in these type of commands.

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+**Prerequisites**
+
+need a "new" install or "clean" debian .
+
+## Step 1 - (Optional) Disable postfix and remove it 
+
+please ignore this step if your system not installed postfix .
+
+```shell
+systemctl disable postfix && /etc/init.d/postfix stop
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: When disabling the service via systemd I'd also point readers to stop it via systemd:
`systemctl stop postfix`

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+apt remove -y --purge postfix*
+apt -y autoremove
+```
+
+## Step 2 - Update os and software
+
+### Step 2.1 - (Optional) set apt to use ipv4 only , because ipv6 is very slow on some region and set ll=ls -al
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: The sentence after the comma could be better suited to be part of the paragraph imho.

Also, when advising readers to create aliases or "temporary - fixes" (like disabling apt from useing v6 resources) - I suppose we should give a little better explanation about that and whether or not these changes are persistent.

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+```shell
+echo 'Acquire::ForceIPv4 "true";' > /etc/apt/apt.conf.d/99force-ipv4
+alias ll='ls -al'
+echo "alias ll='ls -al'" >> /root/.bashrc
+```
+
+### Step 2.2 - update os and software
+
+```shell
+echo "deb http://ftp.de.debian.org/debian stretch main contrib non-free" >/etc/apt/sources.list
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: careful with the arrows! :)

You also seem to be doing a bit more than just "updating the OS". Give our readers a quick insight or a Link to furhter information about what you're doing here.

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+echo "deb http://ftp.de.debian.org/debian stretch-updates main contrib non-free" >>/etc/apt/sources.list
+echo "deb http://security.debian.org stretch/updates main contrib non-free" >>/etc/apt/sources.list
+echo "deb-src http://ftp.de.debian.org/debian stretch main contrib non-free" >>/etc/apt/sources.list
+echo "deb-src http://ftp.de.debian.org/debian stretch-updates main contrib non-free" >>/etc/apt/sources.list
+echo "deb-src http://security.debian.org stretch/updates main contrib non-free" >>/etc/apt/sources.list
+apt-get update && apt-get upgrade -y && apt-get -u dist-upgrade -y
+```
+
+### Step 2.3 - (Optional) Optimize tcp for more-connection
+
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: Also here. You are advising a possibly inexperienced user to do conduct persistent changes to their operating system. Please leave them some more information :)

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+ok , the site is running now ! you can send files use ftp and visit the web site .
+
+
+
+
+
+
+
+
+##### License: MIT
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: License Block is not needed imho - this whole project underlies the MIT - License.

**Comment by eaglepride** [comment]
Thank you, you are very professional and rigorous. I re-uploaded the file

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md (hunk 8 above)
Comment (n-se) [reply tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: This is wrong, the license block **is** needed, see [#66-473822890](https://github.com/hetzneronline/community-content/pull/66#issuecomment-473822890)

Comment (blanky0230) [reply tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: oof. My bad :man_facepalming: @eaglepride

**Comment by eaglepride** [comment]
thanks, it's ok now

File: tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md [hunk 9]
```diff
@@ -169,7 +169,7 @@ systemctl restart nginx
 systemctl restart php7.3-fpm
 ```
 
-ok , the site is running now ! you can send files use ftp and visit the web site.
+ok, the site is running now ! you can send files use ftp and visit the web site.
 
 
-##### License: MIT
\ No newline at end of file
+##### License: MIT
```
Comment (blanky0230) [inline tutorials/Install-the-latest-version-of-php-nginx-and-MariaDB-ProFTPD-(LEMP)-server-on-Debian-9.x/01.en.md]: ``

**Comment by blanky0230** [comment]
Looks good to me :)

**Comment by eaglepride** [comment]
*^_^*
